### PR TITLE
Store URNs of MaintainableArtefacts and test

### PR DESCRIPTION
**Pull request by khaeru**
This adds two tests, and one line of code: https://github.com/dr-leo/pandaSDMX/blob/f53ac118e69c888c4b6d2c1dde47c885b4d6b1de/pandasdmx/reader/sdmxml.py#L469-L470

If I *comment out* that line of code and run the tests (`pytest -k dsd`):
- `TestECB_EXR1.test_uri` **passes**, i.e. the code was already successfully storing URIs.
- `Test_ESTAT_dsd_apro_mk_cola.test_urn` **fails**, i.e. the test confirms that the URN was not being parsed, per #154.

The specimen used by TestECB_EXR1 is the only one in the entire collection of pandasdmx/tests/data/ that contains an XML URI attribute. @dr-leo your title for #154 says that "URN and URI attributes are None"; but the URI *should* be None if it doesn't appear in a message. To be clear, did you see an positive case where a URI *was* in the XML, but did *not* appear on the parsed object?

If so, please describe so it can be reproduced/checked here. If not, then this code resolves the issue.

### CODE REVIEW

File: pandasdmx/reader/sdmxml.py [hunk 1]
```diff
@@ -466,6 +466,9 @@ def _maintained(self, cls=None, id=None, urn=None, **kwargs):
             cls = get_class(match['package'], match['class'])
             id = match['id']
 
+            # Re-add the URN to the kwargs
+            kwargs['urn'] = urn
+
         key = (cls.__name__, id) if isclass(cls) else (cls, id)
```
Comment (dr-leo) [inline pandasdmx/reader/sdmxml.py:473]: This comments is about line 484: 
```python
assert kwargs.pop('is_external_reference', True)
```

You know by now my pop-aversion. The check is wierd. It raises and error if and only if 'is_external_reference' is in kwargs and evaluates to False. If it is not in kwargs, the assertion passes. The correct assert condition, in my view is:

`assert kwargs['is_external_reference']`

This checks for the existence of the key incidentally, and whether it is True as it should be. If you are concerned with a KeyError, use kwarts.get() instead. Upon instantiation in line 485, ìs_external_reference` is in the kwargs so you don't have to pass it explicitely.

**Comment by dr-leo** [comment]
On the URI question: I cannot tell whether the test data contains a structure message with uri attributes. - I had naively assumed that an URI is always present and indeed a valid URL suitable to request the referenced artefact. But based on one of your comments (I forgot where), this may be too good to be true. So let's leave the uri aspect aside for now and merge this PR once you've agreed to do away with the pop thing ;-)

File: pandasdmx/reader/sdmxml.py (hunk 1 above)
Comment (khaeru) [reply pandasdmx/reader/sdmxml.py:473]: > It raises [an] error if and only if 'is_external_reference' is in kwargs and evaluates to False.

Correct! If the code gets to this point under those conditions, something very weird *is* going on with the message being parsed, and we can't continue.

This XML attribute is more frequently absent then present, so we can't simply lookup. But you remind me that dicts also have [setdefault()](https://docs.python.org/3/library/stdtypes.html#dict.setdefault), which we can use instead while also shortening the code.

**Comment by khaeru** [comment]
> See my line comment at line 474 in sdmxml.py, which relates to line 484, 485 though. No big deal, but a little climpsy.

Thanks, see reply and 35dd7de for the change.

Tests pass, so I will merge.

**Pull request merged**